### PR TITLE
docs: update api specs according to guildelines

### PR DESCRIPTION
**Pull request by marekswiecznik**
Issues: MEN-655

Besides style/layout:

* Removed `DELETE /deployments/{id}` (not implemented)

@michaelatmender @maciejmrowiec

### CODE REVIEW

File: docs/integrations_api.yml [hunk 1]
```diff
@@ -61,92 +64,58 @@ paths:
...
           $ref: "#/responses/InternalServerError"
 
     post:
-      summary: Create deployment
+      summary: Create a deployment
       description: |
         Deploy software to specified devices. Image is auto assigned to the
         device from all available images based on artifact name and device type.
 
         NOTE: Because of lack of inventory system, service assumes hardcoded
```
Comment (maciejmrowiec) [inline docs/integrations_api.yml]: Outdated section (remove) - have been fully integrated with inventory.

Can add something like (more less):

```
Devices for which there are no compatible artefacts to be installed are considered finished successfully as well as receive status of `noimage`.
```

File: docs/integrations_api.yml [hunk 2]
```diff
@@ -282,74 +183,48 @@ paths:
...
           $ref: "#/responses/NotFoundError"
         500:
           $ref: "#/responses/InternalServerError"
 
   /images:
     get:
-      summary: List firmware images
+      summary: List known firmware images
+      description: |
+        Returns a paged collection of all firmware images.
```
Comment (maciejmrowiec) [inline docs/integrations_api.yml]: It's not paged

File: docs/integrations_api.yml [hunk 3]
```diff
@@ -374,10 +249,9 @@ paths:
           $ref: "#/responses/InternalServerError"
 
     post:
-      summary: Create firmware image
+      summary: Create a firmware image
```
Comment (maciejmrowiec) [inline docs/integrations_api.yml]: I see we consistently use firmware across this doc, but we already switched to mender artefacts, wondering if we should adjust taxonomy here somethow? (not a blocker, just general remark to figure out) @michaelatmender

File: docs/integrations_api.yml [hunk 4]
```diff
@@ -456,86 +333,207 @@ paths:
           $ref: "#/responses/InternalServerError"
 
     delete:
-      summary: Remove firmware image
+      summary: Delete the firmware image
+      description: |
+        Deletes the firmware image from file and images storage.
```
Comment (maciejmrowiec) [inline docs/integrations_api.yml]: We probably should mention that images used in deployment in progress can't be deleted until deployment finishes (they get locked for that time)

**Comment by marekswiecznik** [comment]
All fixed (except firmawre -> mender artefact, waiting for confirmation), @maciejmrowiec thanks for review.

File: docs/integrations_api.yml [hunk 5]
```diff
@@ -225,51 +155,21 @@ paths:
...
         500:
           $ref: "#/responses/InternalServerError"
 
-  '/deployments/{deployment_id}/devices':
+  /deployments/{deployment_id}/devices:
     get:
       summary: List devices of a deployment
       description: |
-        List deployment status for each assigned device
+        Returns a paged collection of a selected deployment's status for each assigned device.
```
Comment (maciejmrowiec) [inline docs/integrations_api.yml]: It's not paged, not sure of anything in this service is paged.

File: docs/integrations_api.yml [hunk 6]
```diff
@@ -1,39 +1,42 @@
...
+      $ref: "#/definitions/Error"
 
 paths:
   /deployments:
     get:
-      summary: Find deployments
+      summary: Find all deployments
       description: |
-        Lookup deployments in the system, including active and history.
+        Returns a paged, filtered collection of deployments in the system,
```
Comment (maciejmrowiec) [inline docs/integrations_api.yml]: Not paged

**Comment by marekswiecznik** [comment]
All fixed

File: docs/integrations_api.yml (hunk 3 above)
Comment (michaelatmender) [inline docs/integrations_api.yml]: @maciejmrowiec I see what you mean - it's fine for now but I think we will change this once "artifacts" taxonomy is decided on

**Pull request merged**